### ISSUE TITLE
Clean up static code analyse findings

### ISSUE THREAD

**Issue by rettichschnidi** [eclipse/wakaama#578]
With PR #570 merged, *someone* should deal with all the findings and either fix or mark them as false positive. Once the count is at zero, we can start breaking the build if new issues get introduced.

# Acceptance Criteria
* Clang static analyzer does not find any issues anymore
* Build breaks if any subsequent commits introduce new issues

**Comment by tuve** [comment]
I had a quick look at this and noticed that it seems to be findings of type "Dereference of null pointer" this i think have been up for discussion before and my recollection of that was that we (in order to keep the footprint down) didn't want to do null checks then a possible null value was caused by a programmer err. How should we handle those findings, or should we deviate from this and just fix them? @sbernard31 @qleisan @rettichschnidi @mlasch

**Comment by rettichschnidi** [comment]
The overhead for checking for NULL pointers is extremely low, I'd prefer do this always. This project is dealing with data coming from the network, so we should err on save side.

**Comment by tuve** [comment]
I agree, yes there is an overhead but, as with coding standards, any possible gain of "running your own" is so minute compared to the convenience of a tool and a standard.

**Comment by sbernard31** [comment]
Again, I let the team decide on this.
https://github.com/eclipse/wakaama/issues/513#issuecomment-763513438 sounds related to this point.

**Comment by rettichschnidi** [comment]
Closing because the Wakaama project will be terminated: https://gitlab.eclipse.org/eclipsefdn/emo-team/emo/-/issues/1191
